apisix/plugins/ip-restriction.lua

```diff
@@ -27,21 +27,31 @@ local lrucache  = core.lrucache.new({
 
 local schema = {
     type = "object",
-    properties = {
-        whitelist = {
-            type = "array",
-            items = {type = "string", anyOf = core.schema.ip_def},
-            minItems = 1
+    oneOf = {
+        {
+            title = "whitelist",
+            properties = {
+                whitelist = {
+                    type = "array",
+                    items = {type = "string", anyOf = core.schema.ip_def},
```

Comment:
```json
{
  "type": "object",
  "oneOf": [
    {
      "title": "whitelist",
      "properties": {
        "whitelist": {
          "title": "",
          "type": "array",
          "items": {
            "anyOf": [
              {
                "title": "Pattern 1",
                "type": "string",
                "pattern": "This is Pattern"
              }
            ]
          },
          "minItems": 1
        }
      },
      "required": ["whitelist"],
      "additionalProperties": false
    }
  ]
}
```

This schema is ok for me.

Reply:
1. We should add type for every item in anyOf;
2. remove string type from items.

Reply:
[Playground](https://rjsf-team.github.io/react-jsonschema-form/#eyJmb3JtRGF0YSI6eyJ3aGl0ZWxpc3QiOlsiMjIyIl0sImJsYWNrbGlzdCI6W251bGxdLCJmaXJzdE5hbWUiOiJDaHVjayIsImxhc3ROYW1lIjoiTm9ycmlzIiwiYWdlIjo3NSwiYmlvIjoiUm91bmRob3VzZSBraWNraW5nIGFzc2VzIHNpbmNlIDE5NDAiLCJwYXNzd29yZCI6Im5vbmVlZCJ9LCJzY2hlbWEiOnsidHlwZSI6Im9iamVjdCIsIm9uZU9mIjpbeyJ0aXRsZSI6IndoaXRlbGlzdCIsInByb3BlcnRpZXMiOnsid2hpdGVsaXN0Ijp7InRpdGxlIjoiIiwidHlwZSI6ImFycmF5IiwiaXRlbXMiOnsiYW55T2YiOlt7InRpdGxlIjoiUGF0dGVybiAxIiwidHlwZSI6InN0cmluZyIsInBhdHRlcm4iOiJUaGlzIGlzIFBhdHRlcm4ifV19LCJtaW5JdGVtcyI6MX19LCJyZXF1aXJlZCI6WyJ3aGl0ZWxpc3QiXSwiYWRkaXRpb25hbFByb3BlcnRpZXMiOmZhbHNlfV19LCJ1aVNjaGVtYSI6eyJmaXJzdE5hbWUiOnsidWk6YXV0b2ZvY3VzIjp0cnVlLCJ1aTplbXB0eVZhbHVlIjoiIiwidWk6YXV0b2NvbXBsZXRlIjoiZmFtaWx5LW5hbWUifSwibGFzdE5hbWUiOnsidWk6ZW1wdHlWYWx1ZSI6IiIsInVpOmF1dG9jb21wbGV0ZSI6ImdpdmVuLW5hbWUifSwiYWdlIjp7InVpOndpZGdldCI6InVwZG93biIsInVpOnRpdGxlIjoiQWdlIG9mIHBlcnNvbiIsInVpOmRlc2NyaXB0aW9uIjoiKGVhcnRoaWFuIHllYXIpIn0sImJpbyI6eyJ1aTp3aWRnZXQiOiJ0ZXh0YXJlYSJ9LCJwYXNzd29yZCI6eyJ1aTp3aWRnZXQiOiJwYXNzd29yZCIsInVpOmhlbHAiOiJIaW50OiBNYWtlIGl0IHN0cm9uZyEifSwiZGF0ZSI6eyJ1aTp3aWRnZXQiOiJhbHQtZGF0ZXRpbWUifSwidGVsZXBob25lIjp7InVpOm9wdGlvbnMiOnsiaW5wdXRUeXBlIjoidGVsIn19fSwidGhlbWUiOiJkZWZhdWx0IiwibGl2ZVNldHRpbmdzIjp7InZhbGlkYXRlIjpmYWxzZSwiZGlzYWJsZSI6ZmFsc2UsIm9taXRFeHRyYURhdGEiOmZhbHNlLCJsaXZlT21pdCI6ZmFsc2V9fQ==)